pkg/patterns/declarative/reconciler.go

```diff
@@ -230,6 +230,25 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 			if _, ok := annotations["addons.k8s.io/ignore"]; ok {
 				log.WithValues("kind", obj.Kind).WithValues("name", obj.Name).Info("Found ignore annotation on object, " +
 					"skipping object")
+
+				// remove labels, avoiding be removed when kubectl.Apply is called with flag `prune`
+				needUpdate := false
+				{ // remove labels from object
+					labels := unstruct.GetLabels()
+					for k := range r.options.labelMaker(ctx, instance) {
+						if _, ok := labels[k]; ok {
+							needUpdate = true
+						}
+						delete(labels, k)
+					}
+					unstruct.SetLabels(labels)
+				}
+				if needUpdate { // update object if it's necessary
+					if _, err := UpdateObjectFromCluster(unstruct, r); err != nil {
+						return objects, fmt.Errorf("remove label for resource with 'addons.k8s.io/ignor' failed, %w", err)
```

Comment:
```suggestion
						return objects, fmt.Errorf("remove label for resource with 'addons.k8s.io/ignore' failed, %w", err)
```
maybe typo?

Reply:
oh, indeed. I will update this PR until we find a right way with this https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/210#issuecomment-1078749278